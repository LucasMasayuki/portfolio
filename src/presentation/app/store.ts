/* eslint-disable import/no-cycle */
import { Action, configureStore, ThunkAction } from '@reduxjs/toolkit';
import { colorModeSlice } from '../stores/color-mode-slice';
import { notificationSlice } from '../stores/notification-slice';

export const store = configureStore({
  reducer: {
    colorMode: colorModeSlice.reducer,
    notification: notificationSlice.reducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
