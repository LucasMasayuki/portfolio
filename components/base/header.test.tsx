import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { describe, expect, it } from 'vitest';
import { store } from '../../src/presentation/app/store';
import Header from './header';

describe.concurrent('Header component', () => {
  describe('when render', () => {
    it('should fill the header with correct components', () => {
      // When rendered
      const { container } = render(
        <Provider store={store}>
          <Header />
        </Provider>
      );

      // Then it should show all content
      expect(container.innerHTML).toMatchSnapshot();
    });
  });
});
