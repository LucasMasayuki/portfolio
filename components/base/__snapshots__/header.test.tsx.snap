// Vitest Snapshot v1

exports[`Header component > when render > should fill the header with correct components 1`] = `"<header class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-189twtb-MuiPaper-root-MuiAppBar-root\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root\\"><div class=\\"MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 css-1x7nvub-MuiTypography-root\\">LOGO</h6> <div class=\\"MuiBox-root css-i9gxme\\"></div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-9ae4v2-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"Brightness4Icon\\"><path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><div class=\\"MuiBox-root css-5nwj3y\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Conta\\" data-mui-internal-clone-element=\\"true\\"><div class=\\"MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-6u9n7p-MuiAvatar-root\\">T</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></header>"`;
